src/sentry/identity/slack/provider.py

```diff
@@ -12,25 +12,46 @@ class SlackIdentityProvider(OAuth2Provider):
 
     # This identity provider is used for authorizing the Slack application
     # through their Bot token (or legacy Workspace Token if enabled) flow.
-    oauth_authorize_url = "https://slack.com/oauth/authorize"
 
     oauth_scopes = ("identity.basic", "identity.email")
 
+    def get_oauth_authorize_url(self):
+        if self.use_wst_app:
+            return "https://slack.com/oauth/authorize"
+        return "https://slack.com/oauth/v2/authorize"
+
+    @property
+    def use_slack_v2(self):
+        # need to import here because of dependency issues
+        # would be nice to make the pipeline class have a use_slack_v2
+        # method but we use the same pipeline class everywhere
```

Comment:
IMO this is preferred to have it here, and makes sense to import like this. 👍

Reply:
Eh...I don't prefer it this way but I don't think it will ever change. I can remove the second part of the comment.